### PR TITLE
Create codeQL-scanning.yaml

### PR DESCRIPTION
**Pull request by anupsv**
## Checks

- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- [ ] I've checked the new test coverage and the coverage percentage didn't drop.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: .github/workflows/codeQL-scanning.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+name: "CodeQL scanning"
+
+on:
+  push:
+    branches:
+      - master
+      - 'release/*'
+  pull_request:
+    branches:
+      - main
```
Comment (jianoaix) [inline .github/workflows/codeQL-scanning.yaml:10]: There is no `main` branch

File: .github/workflows/codeQL-scanning.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      - 'operators/**'
+      - 'retriever/**'
+      - 'disperser/**'
+      - 'core/**'
+      - 'common/**'
+      - 'api/**'
+      - '.github/codeql/**'
+      - '.github/workflows/codeql-analysis.yml'
+  schedule:
+    - cron: '0 9 * * *'
```
Comment (jianoaix) [inline .github/workflows/codeQL-scanning.yaml:23]: It's triggered already on push, what this cron test adds?

File: .github/workflows/codeQL-scanning.yaml (hunk 1 above)
Comment (anupsv) [reply .github/workflows/codeQL-scanning.yaml:10]: we should probably change to main. `master` seems deprecated as the default name.

File: .github/workflows/codeQL-scanning.yaml (hunk 2 above)
Comment (anupsv) [reply .github/workflows/codeQL-scanning.yaml:23]: adds a cadence of running the codeql scanner everyday at 9 am. Helps keep a regular non PR triggered scan of the code. Idea is to use this for longer and slower codeql scans. Will be adding that soon-ish.

**Review by ian-shim — APPROVED** [review]

**Review by jianoaix — APPROVED** [review]

**Pull request merged**